src/analyzer/protocol/ntp/NTP.cc

```diff
@@ -1,114 +1,38 @@
-// See the file "COPYING" in the main distribution directory for copyright.
-
-#include "zeek-config.h"
-
-#include "NetVar.h"
 #include "NTP.h"
-#include "Sessions.h"
-#include "Event.h"
+
+#include "Reporter.h"
 
 #include "events.bif.h"
 
-using namespace analyzer::ntp;
+using namespace analyzer::NTP;
 
-NTP_Analyzer::NTP_Analyzer(Connection* conn)
-	: Analyzer("NTP", conn)
+NTP_Analyzer::NTP_Analyzer(Connection* c)
+	: analyzer::Analyzer("NTP", c)
 	{
-	ADD_ANALYZER_TIMER(&NTP_Analyzer::ExpireTimer,
-				network_time + ntp_session_timeout, 1,
-				TIMER_NTP_EXPIRE);
+	interp = new binpac::NTP::NTP_Conn(this);
 	}
 
-void NTP_Analyzer::Done()
+NTP_Analyzer::~NTP_Analyzer()
 	{
-	Analyzer::Done();
-	Event(udp_session_done);
-	}
-
-void NTP_Analyzer::DeliverPacket(int len, const u_char* data, bool is_orig, uint64 seq, const IP_Hdr* ip, int caplen)
-	{
-	Analyzer::DeliverPacket(len, data, is_orig, seq, ip, caplen);
-
-	// Actually we could just get rid of the Request/Reply and simply use
-	// the code of Message().  But for now we use it as an example of how
-	// to convert an old-style UDP analyzer.
-	if ( is_orig )
-		Request(data, len);
-	else
-		Reply(data, len);
+	delete interp;
 	}
 
-int NTP_Analyzer::Request(const u_char* data, int len)
+void NTP_Analyzer::Done()
```

Comment:
If all you're doing is calling the parent method in this method, you don't need to override it. The child version can be removed.

Reply:
correct. I removed the method